### ISSUE TITLE
Mob dimlets are limited to vanilla mobs

### ISSUE THREAD

**Issue by LemonatedCat** [McJty/RFToolsDimensions#306]
As of right now there seems to be no way to add custom mob dimlets, as the 1.12.2 release uses a hardcoded list of allowed mobs, if something is not in that list, it gets excluded. I Believe the 1.10.2 version did not do this though, if i get a response from the mod author i also may make a pull request to add it back. If the author does not care though, i won't do so.

**Comment by McJty** [comment]
It's not hardcoded but you have to remember to also edit the syringe settings for the rftools spawner as the mob dimlets use that system too

**Comment by LemonatedCat** [comment]
Oh wtf, why is it only using base vanilla mobs then, that is strange. The syringes exist, but they mostly do not work for this. Weird.

**Comment by LemonatedCat** [comment]
I'm a bit confused, how exactly would this work? The syringes exist, but the spawners show nullRF for the non standard mobs and some vanilla mobs aswell. Accordingly they also do not get shown in RFTools Dimensions (Wither skeletons are an exception, they do not get listed in the dimlet workbench even though they show up in the spawner)


I cannot find any syringe settings in any config file

**Comment by McJty** [comment]
Is they spawner shows nullRF then the syringes are not configured correctly. I no longer have a 1.12 mod environment set up but rftools.cfg should contain all configuration for all the mobs that can be used by the spawner

**Comment by LemonatedCat** [comment]
Yeah, i see now. Still confused why it worked with no extra configuration in 1.10.2 though. It's rather messy to add these entries aswell.

**Comment by LemonatedCat** [comment]
I see now aswell, it looks like it is doable within rftools dimensions itself in the config file, no?
```

for (int i = 0 ; i < list.tagCount() ; i++) {
            NBTTagCompound tc = list.getCompoundTagAt(i);
            String className = tc.getString("class");
            int chance = tc.getInteger("chance");
            int minGroup = tc.getInteger("minGroup");
            int maxGroup = tc.getInteger("maxGroup");
            int maxLoaded = tc.getInteger("maxLoaded");
            Class<? extends EntityLiving> c = null;
            try {
                c = (Class<? extends EntityLiving>) Class.forName(className);
            } catch (ClassNotFoundException e) {
                throw new RuntimeException(e);
            }
            MobDescriptor mob = new MobDescriptor(c, chance, minGroup, maxGroup, maxLoaded);
            extraMobs.add(mob);
        }

        String ds = tagCompound.getString("dimensionTypes");
        dimensionTypes = StringUtils.split(ds, ",");
        if (dimensionTypes == null) {
            dimensionTypes = new String[0];
        }
```

**Comment by LemonatedCat** [comment]
Well i guess not, that's annoying, but hey guess i'll have to deal with it. Maybe the rftools dimensions config should be used as a fallback for these settings, because you cannot set defaults inside the normal rftools for the spawner configs.

**Comment by LemonatedCat** [comment]
This also does not explain why Wither skeleton dimlets also refuse to work, because their syringes are set up properly, and they seem to be using a different id aswell in this case. They should be using the fallback "mob" filter, no?
